tests/system/test_agent_keep_alive/test_agent_keep_alive.py

```diff
@@ -0,0 +1,339 @@
+from wazuh_testing.tools.file import read_yaml
+from wazuh_testing.tools.monitoring import FileMonitor, HostMonitor
+from wazuh_testing.tools.system import HostManager
+
+GLOBAL_DB_PATH = f"{WAZUH_PATH}/queue/db/global.db"
+
+# Hosts
+testinfra_hosts = ["wazuh-manager", "wazuh-agent1"]
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+                              'provisioning', 'basic_environment', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+messages_path = os.path.join(local_path, 'data/messages.yml')
+tmp_path = os.path.join(local_path, 'tmp')
+agent_conf_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..',
+                               'provisioning', 'basic_environment', 'roles', 'agent-role', 'files', 'ossec.conf')
+manager_conf_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data/config.yml')
+test_cases_yaml = read_yaml(os.path.join(local_path, 'data/test_agent_keep_alive_cases.yml'))
+
+wait_agent_start = 70
+network = {}
+
+
+def callback_sending_keep_alive(line):
+    global info
+    match = re.match(r'.*Sending keep alive.*', line)
+    if match:
+        keep_alive = match.group(0)
+        info = keep_alive.split('#!-')[1]
+        return keep_alive
+
+
+def callback_sending_merged(line):
+    global merged
+    match = re.match(r'.*merged.mg*', line)
+    if match:
+        merged = match.group(0)
+        return merged
+
+
+def callback_sending_agent_ip(line):
+    global agent_ip
+    match = re.match(r'.*agent_ip.*', line)
+    if match:
+        agent_ip = match.group(0)
+        return agent_ip
+
+
+def callback_reading_keep_alive(line):
+    global reading
+    match = re.match(r'.*inserting.*', line)
+    if match:
+        reading = match.group(0)
+        reading = reading.split("'")[1]
+        return match.group(0)
+
+
+def callback_reading_merged(line):
+    global reading_merged
+    match = re.match(r'.*merged.mg*', line)
+    if match:
+        reading_merged = match.group(0)
+        return reading_merged
+
+
+def callback_reading_agent_ip(line):
+    global reading_agent_ip
+    match = re.match(r'.*agent_ip.*', line)
+    if match:
+        reading_agent_ip = match.group(0)
+        return reading_agent_ip
+
+
+def callback_inserting_keep_alive(line):
+    global inserting
+    match = re.match(r'.*reading.*', line)
+    if match:
+        inserting = match.group(0)
+        inserting = inserting.split("'")[1]
+        return match.group(0)
+
+
+def callback_inserting_merged(line):
+    global inserting_merged
+    match = re.match(r'.*merged.mg*', line)
+    if match:
+        inserting_merged = match.group(0)
+        return inserting_merged
+
+
+def callback_inserting_agent_ip(line):
+    global inserting_agent_ip
+    match = re.match(r'.*agent_ip.*', line)
+    if match:
+        inserting_agent_ip = match.group(0)
+        return inserting_agent_ip
```

Comment:
These callbacks functions can be refactored. Also, remove global variables and use the return value.

Reply:
Done in commit fdbd7de and c6bbdfd